### ISSUE TITLE
Unable to create Service Monitor object if all namespaces are watched

### ISSUE THREAD

**Issue by sarahhenkens** [ricoberger/vault-secrets-operator#37]
If no specific namespaces are defined, the value for the "watchNamespaces" === `""`:
https://github.com/ricoberger/vault-secrets-operator/blob/54b0db8d33403680eaaa93f769fee83e964c8156/cmd/manager/main.go#L120
```go
namespace, err := k8sutil.GetWatchNamespace()

...somewhat later...

if namespace == "" {
	mgrOpts.Namespace = namespace
	log.Info("Watch all namespaces")
} else {
```

This `namespace` variable eventually gets used to create its service monitor:
```go
// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
// necessary to configure Prometheus to scrape metrics from this operator.
services := []*v1.Service{service}
_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
```

In our example, `namespace` equals an empty string and this is causing the following error:

> an empty namespace may not be set during creation

```
{"level":"info","ts":1588909453.9949079,"logger":"cmd","msg":"Registering Components."}
{"level":"info","ts":1588909453.995184,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":
"vaultsecret-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588909453.9953754,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller"
:"vaultsecret-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588909455.9308386,"logger":"metrics","msg":"Metrics Service object created","Service.Name":"vault-se
crets-operator-metrics","Service.Namespace":"vault"}
{"level":"error","ts":1588909456.8837671,"logger":"cmd","msg":"Could not create ServiceMonitor object","error":"an empty n
amespace may not be set during creation","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/home/runner/go/pkg/m
od/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nmain.main\n\tvault-secrets-operator/cmd/manager/main.go:196\nruntime.main\n
\t/opt/hostedtoolcache/go/1.13.10/x64/src/runtime/proc.go:203"}
```

### Expected behavior:

For the service monitor to be created in the namespace of the operator itself.

### Ideal behavior:

Do not let the go application code create these kubernetes resources and let end-users manage those themselves. Only expose the metrics port on the go application. The way the operator is built right now, we're unable to add any additional labels/config/annotations to the service monitor object itself. Making it unusuable in non-standard setups of prometheus.

**Comment by ricoberger** [comment]
Hi @sarahhenkens first of all: thanks for the detailed description. I'm completely with you and created a PR (#38) for this. When I tested this I will create a new release.

**Comment by ricoberger** [comment]
I published version [1.6.0](https://github.com/ricoberger/vault-secrets-operator/releases/tag/1.6.0) where the `ServiceMonitor` is not created anymore.

If you have further questions or suggestions feel free to reopen the issue or open a new one.

**Comment by sarahhenkens** [comment]
Thank you for the super quick fix and release 🙌
